core/src/main/java/org/apache/calcite/plan/RelOptRuleCall.java

```diff
@@ -201,13 +201,18 @@ public RelOptPlanner getPlanner() {
    * @return true iff rule should be excluded
    */
   public boolean isRuleExcluded() {
-    if (!(rels[0] instanceof Hintable)) {
-      return false;
+    boolean excluded = false;
```

Comment:
You don't need the `excluded` variable at all it seems, remove this line and replace the latest return statement with `return false;` (sorry but multi-line comment is broken in Github UI in this case so I can't use code suggestion).

Reply:
Good catch! Removed